tests/karma/unit/controllers/edit-user.js

```diff
@@ -271,12 +271,93 @@ describe('EditUserCtrl controller', function() {
       chai.expect(scope.errors).not.to.have.property('password');
     });
 
+    it('must have associated place if user type is restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'district-manager';
+      scope.editUserModel.facility_id = undefined;
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).to.have.property('facility_id');
+    });
+
+    it('must have associated contact if user type is restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'district-manager';
+      scope.editUserModel.contact_id = undefined;
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).to.have.property('contact_id');
+    });
+
+    it('must have associated place and contact if user type is restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'district-manager';
+      scope.editUserModel.facility_id = undefined;
+      scope.editUserModel.contact_id = undefined;
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).to.have.property('facility_id');
+      chai.expect(scope.errors).to.have.property('contact_id');
+    });
+
+    it('doesn\'t need associated place and contact if user type is not restricted user', function() {
+      UpdateUser.returns(KarmaUtils.mockPromise());
+      dbQuery.returns(KarmaUtils.mockPromise());
+      createController();
+
+      scope.editUserModel.type = 'some-other-type';
+
+      // when
+      scope.editUser();
+
+      // expect
+      chai.expect(scope.errors).not.to.have.property('facility_id');
+      chai.expect(scope.errors).not.to.have.property('contact_id');
+    });
+
+
+    var jQuery;
+    // If you use this, don't forget to reset: you're mocking out
+    // jQuery for all unit tests.
+    var mockOutjQueryFormFields = function() {
+      jQuery = window.$;
+      window.$ = sinon.stub();
+      window.$.withArgs('#edit-user-profile [name=contact]').returns(
+        {val: function(){ return model.contact_id; }});
```

Comment:
Tying jQuery to the angular model like this seems odd.  I've tried to find other karma tests that we've written which mock jquery at all but haven't been successful.